File: animation-backend/src/main/java/com/example/boot/Entity/Animation.java
package com.example.boot.Entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDateTime;


@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class Animation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    private String title;
    private String content;
    private String photo;
    private String genre;
    private String dayOfWeek;
    private double averageRating;
    private String uploaded;
    private Long viewCount;
    private String viewed;
    private LocalDateTime viewedTime;
    private String favorite;

    private String anidelete = "n";

    private LocalDateTime date = LocalDateTime.now();

}

File: animation-frontend/src/model/Animation.ts
export type AnidataTs = {
  id: number;
  title: string;
  photo: string;
  content: string;
  date: Date;
  dayOfWeek: string;
  genre: string;
  viewCount: number;
  averageRating: number;
};

export type UserViewAnidataTs = Omit<
  AnidataTs,
  'content' | 'genre' | 'date' | 'dayOfWeek' | 'viewCount'
>;

export type GenreArray = {
  genre: string[];
};
